internal/pkg/service/buffer/store/file.go

```diff
@@ -253,9 +253,23 @@ func (s *Store) setFileStateOp(ctx context.Context, now time.Time, file *model.F
 	}
 
 	// Atomically swap keys in the transaction
+	alreadyInState := false
 	ops := []op.Op{
-		s.schema.Files().InState(from).ByKey(file.FileKey).DeleteIfExists(),
-		s.schema.Files().InState(to).ByKey(file.FileKey).PutIfNotExists(clone),
+		s.schema.Files().InState(to).ByKey(file.FileKey).PutIfNotExists(clone).WithOnResultOrErr(func(ok bool) error {
+			alreadyInState = !ok
+			if !ok {
+				file.State = to
+				return errors.Errorf(`file "%s" is already in the "%s" state`, file.FileKey, to)
+			}
+			return nil
+		}),
+		s.schema.Files().InState(from).ByKey(file.FileKey).DeleteIfExists().WithOnResultOrErr(func(ok bool) error {
+			if !ok && !alreadyInState {
+				return errors.Errorf(`file "%s" not found in the "%s" state`, file.FileKey, file.State)
+			}
+			return nil
+
+		}),
```

Comment:
Tu chybali tieto chyby, mal som v preklep v teste a nevedel som najst problem, v `store/slice.go` to mame ale tu sa na to nejak zabudlo.